### PR TITLE
new Angular 6 library for Appworks

### PR DESCRIPTION
**Pull request by zhe-db**
new Angular 6 library for Appworks

### CODE REVIEW

**Comment by jasonaibrahim** [comment]
this is great @z687wang 
i have a few notes

* i think we can provide a less clunky api by extracting each plugin out into its own service. for example:
```typescript

@Injectable({
  providedIn: 'root'
})
export class CameraService { 
...
}

@Injectable({
  providedIn: 'root'
})
export class AuthService { 
...
}
```

Then in a component we can have a more focused api:

```typescript
@Component({
  selector: 'app-my-component',
  template: `
    <p>
      ng-appworksjs works!
    </p>
  `,
  styles: []
})
export class MyComponent implements OnInit {
  constructor(private auth: AuthService, private camera: CameraService) {
    ...
  }
}
```
* i would like to see the services provided in the module itself. basically I shouldnt need to import the module, _and_ declare a provider. see https://github.com/ngx-translate/core/blob/master/projects/ngx-translate/core/src/public_api.ts for an example
* im seeing a duplicate file `ng-appworks.services.ts` and `ng-appworks.service.ts`
* on a clean install i get a test failure `Cannot find module 'appworks-js' from 'ng-appworks.component.ts'`. perhaps appworks-js should be added to the main project's package.json, or declared as regular `dependency` in `package.json`
* even with appworks-js added as a dependency, the test suite wont run. 
```
 FAIL  src/lib/ng-appworks.component.spec.ts
  ● Test suite failed to run

    /Users/jasonaibrahim/Desktop/angular-appworks/projects/ng-appworks/node_modules/appworks-js/src/plugins/accelerometer/index.ts:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){export * from './accelerometer';
                                                                                             ^^^^^^
    
    SyntaxError: Unexpected token export
      
      at ScriptTransformer._transformAndBuildScript (node_modules/jest-runtime/build/script_transformer.js:316:17)
      at Object.<anonymous> (node_modules/appworks-js/src/index.js:3:15)
```

This could have to do with how jest is configured

* last but not least, I would love to have a `NgAppworksTestingModule` for use in development/testing. Basically it would provide the same api as the underlying plugins, but no need to actually reach out to the actual appworks-js function calls. this doesnt need to be done for this PR though. I may piggy back off of what you've done here to create it.

**Review by jasonaibrahim — CHANGES_REQUESTED** [review]
changes requested in prior comment

**Comment by zhe-db** [comment]
Thanks for the code review, @jasonaibrahim , I will rework this one as soon as I finish up the desktop client.

File: projects/ng-appworks/src/lib/Appmanager.service.ts [hunk 1]
```diff
@@ -5,19 +5,14 @@ import { Observable, Observer } from 'rxjs';
...
-        this.onChange = new Observable((obs) => {
-            this.AWAppManager = new AWAppManager((data) => obs.next(data), (err) => obs.error(err));
-        });
-    }
-
-    init(ob: Observer<any>) {
-        this.onChange.subscribe(ob);
     }
 
     colseActiveApp() {
```
Comment (jasonaibrahim) [inline projects/ng-appworks/src/lib/Appmanager.service.ts]: we have a bit of a typo here

**Review by jasonaibrahim — COMMENTED** [review]
this is looking good. the only thing im still thinking about is binding the plugin object instances to the service itself, e.g. `this.AWAccelerometer = new AWAccelerometer(...)`

there doesnt seem to be much use for the binding except when needed to, for example, stop a listener from emitting data. it seems like we could even break it in certain situations, for example, in AWAuthService calling `otdsssoticket()` and then `authenticate()` immediately after would cause a new AWAuth object to get bound to `this`

I would reconsider how youre constructing and returning these objects and try to construct an example. were getting closer to the final product though its looking good.

**Pull request closed without merging**